core/Period/Factory.php

```diff
@@ -55,6 +56,17 @@ public static function build($period, $date, $timezone = 'UTC')
                 return new Year($date);
                 break;
         }
+
+        $customPeriodFactories = StaticContainer::get('period.custom.factories');
```

Comment:
It could still be removed if a plugin sets it to an empty array or uses DI\decorate to remove it. That said it would be hard for a plugin developer to not notice if they removed it, so perhaps it's not really an issue. I'll put them in CoreHome, seems like the better path.